node/comparisons.ts

```diff
@@ -0,0 +1,619 @@
+        }
+        if (keys1.length !== Object.keys(obj2).length) {
+          return false;
+        }
+        return true;
+      }
+    }
+  }
+
+  // Expensive test
+  for (i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    if (!isDeepEqual(obj1[key], obj2[key], strict, memos)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function setEquiv(set1: any, set2: any, strict: boolean, memos: Memo): boolean {
+  let set = null;
+  for (const item of set1) {
+    if (typeof item === "object" && item !== null) {
+      if (set === null) {
+        // What is SafeSet from primordials?
+        // set = new SafeSet();
+        set = new Set();
+      }
+      set.add(item);
+    } else if (!set2.has(item)) {
+      if (strict) return false;
+
+      // Since we do not need to handle non-strict case
+      // if (!setMightHaveLoosePrim(set1, set2, item)) {
+      //   return false;
+      // }
+
+      // if (set === null) {
+      //   set = new Set();
+      // }
+      // set.add(item);
+    }
+  }
+
+  if (set !== null) {
+    for (const item of set2) {
+      if (typeof item === "object" && item !== null) {
+        if (!setHasEqualElement(set, item, strict, memos)) return false;
+      } else if (
+        !strict &&
+        !set1.has(item) &&
+        !setHasEqualElement(set, item, strict, memos)
+      ) {
+        return false;
+      }
+    }
+    return set.size === 0;
+  }
+
+  return true;
+}
+
+// TODO Implementation of non-strict cases is pending
+function mapEquiv(map1: any, map2: any, strict: boolean, memos: Memo): boolean {
+  let set = null;
+
+  for (const { 0: key, 1: item1 } of map1) {
+    if (typeof key === "object" && key !== null) {
+      if (set === null) {
+        set = new Set();
+      }
+      set.add(key);
+    } else {
+      const item2 = map2.get(key);
+      if (
+        (item2 === undefined && !map2.has(key)) ||
+        !isDeepEqual(item1, item2, strict, memos)
+      ) {
+        if (strict) return false;
+        // if (!mapMightHaveLoosePrim(map1, map2, key, item1, memos)) return false;
+        // if (set === null) {
+        //   set = new Set();
+        // }
+        // set.add(key);
+      }
+    }
+  }
+
+  if (set !== null) {
+    for (const { 0: key, 1: item } of map2) {
+      if (typeof key === "object" && key !== null) {
+        if (!mapHasEqualEntry(set, map1, key, item, strict, memos)) {
+          return false;
+        }
+      }
+      // else if (
+      //   !strict &&
+      //   // (!map1.has(key) || !isDeepEqual(map1.get(key), item, false, memos)) &&
+      //   // !mapHasEqualEntry(set, map1, key, item, false, memos)
+      // ) {
+      //   return false;
+      // }
+    }
+    return set.size === 0;
+  }
+
+  return true;
+}
+
+// function setMightHaveLoosePrim(set1, set2, primitive) {
+//   const altValue = findLooseMatchingPrimitives(primitive);
+//   if (altValue != null) return altValue;
+
+//   return set2.has(altValue) && !set1.has(altValue);
+// }
```

Comment:
ditto. please leave some TODO comment about this commented-out block, or remove them